widget/embedded/src/hooks/useFetchCustomToken.ts

```diff
@@ -0,0 +1,107 @@
+import type { Token } from 'rango-sdk';
+
+import { i18n } from '@lingui/core';
+import { useState } from 'react';
+
+import { useAppStore } from '../store/AppStore';
+import { getConfig } from '../utils/configs';
+
+type ErrorType = {
+  title: string;
+  message: string;
+  type: 'TOKEN_ERROR' | 'NETWORK_ERROR';
+};
+
+interface UseFetchCustomToken {
+  fetchCustomToken: ({
+    blockchain,
+    tokenAddress,
+  }: {
+    blockchain: string;
+    tokenAddress: string;
+  }) => Promise<Token>;
+  loading: boolean;
+  error?: ErrorType;
+}
+
+export function useFetchCustomToken(): UseFetchCustomToken {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<undefined | ErrorType>(undefined);
+  const { findToken, customTokens } = useAppStore();
+
+  const handleTokenIsDuplicateError = () => {
```

Comment:
Instead of this approach you could do this:

```
function produceErrorMessage(type: "duplicated" | "not-found" | ...){
switch(type){
case "...":
 return {
   title: ...,
   message: ..,
   type: ...
 }
break;
}
}
```

and usage will be like: 

```
      if (isDuplicate) {
        return setError(produceErrorMessage("duplicated"));
      }
```

Reply:
fixed